src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -505,6 +511,37 @@ export class DataViewsService {
     return fields;
   };
 
+  /**
+   * Get existing index pattern list by providing string array index pattern list.
+   * @param indices index pattern list
+   * @returns index pattern list
+   */
+  getExistingIndices = async (indices: string[]): Promise<string[]> =>
+    await Promise.all(
```

Comment:
Is this going to fire off a lot of requests all at once? I'm concerned that a very large number could crash kibana. Does `map` run sequentially or in parallel? Another approach to consider - https://stackoverflow.com/questions/34535685/limit-number-of-requests-at-a-time-with-rxjs